libraries/AC_WPNav/AC_Circle.cpp

```diff
@@ -227,24 +228,24 @@ bool AC_Circle::update_ms(float climb_rate_ms)
     // calculate z-axis target
     float target_z_m;
     if (_is_terrain_alt) {
-        target_z_m = _center_neu_m.z + terrain_u_m;
+        target_z_m = _center_ned_m.z + terrain_u_m;
     } else {
         target_z_m = _pos_control.get_pos_desired_U_m();
     }
 
     // Construct target position centered on the circle center
-    Vector3p target_neu_m {
-        _center_neu_m.x,
-        _center_neu_m.y,
+    Vector3p target_ned_m {
+        _center_ned_m.x,
+        _center_ned_m.y,
         target_z_m
```

Comment:
I think "target_z_m" may be in the "up" frame in which case it needs a negative here